Tests/Marketplace/marketplace_services.py

```diff
@@ -2365,6 +2373,182 @@ def is_unified_integration(self, file_path: str):
+        does not have a BC text - concat the whole RN). In the future this might be needed to re-thought.
+        Args:
+            release_notes_dir (str): RN dir path.
+            text_of_bc_versions ([List[str]): List of text of BC versions with text.
+            bc_versions_without_text ([List[str]): List of BC versions without text.
+
+        Returns:
+            (str): Text for BC entry.
+        """
+        bc_with_text_str = '\n'.join(text_of_bc_versions)
+        rn_file_names_without_text = [f'''{bc_version.replace('.', '_')}.md''' for
+                                      bc_version in bc_versions_without_text]
+        other_rn_text: str = self._get_release_notes_concat_str(release_notes_dir, rn_file_names_without_text)
+        if not other_rn_text:
+            logging.error('No RN text, although text was expected to be found for versions'
+                          f' {rn_file_names_without_text}.')
+        return f'{bc_with_text_str}{other_rn_text}'
+
+    @staticmethod
+    def _get_release_notes_concat_str(release_notes_dir: str, rn_file_names: List[str]) -> str:
+        """
+        Concat all RN data found in given `rn_file_names`.
+        Args:
+            release_notes_dir (str): RN dir path.
+            rn_file_names (List[str]): List of all RN files to concat their data.
+
+        Returns:
+            (str): Concat RN data
+        """
+        concat_str: str = ''
+        for rn_file_name in rn_file_names:
+            rn_file_path = os.path.join(release_notes_dir, rn_file_name)
+            with open(rn_file_path, 'r') as f:
+                # Will make the concat string start with new line on purpose.
+                concat_str = f'{concat_str}\n{f.read()}'
+        return concat_str
+
+    @staticmethod
+    def _split_bc_versions_with_and_without_text(bc_versions: Dict[str, Optional[str]]) -> Tuple[List[str], List[str]]:
+        """
+        Splits BCs to tuple of BCs text of BCs containing text, and BCs versions that do not contain BC text.
+        Args:
+            bc_versions (Dict[str, Optional[str]): BC versions mapped to text if exists.
+
+        Returns:
+            (Tuple[List[str], List[str]]): (text of bc versions with text, bc_versions_without_text).
+        """
+        text_of_bc_versions_with_tests: List[str] = []
+        bc_versions_without_text: List[str] = []
+        for bc_version, bc_text in bc_versions.items():
+            if bc_text:
+                text_of_bc_versions_with_tests.append(bc_text)
+            else:
+                bc_versions_without_text.append(bc_version)
+        return text_of_bc_versions_with_tests, bc_versions_without_text
+
+    @staticmethod
+    def _breaking_changes_versions_to_text(release_notes_dir: str) -> Dict[str, Optional[str]]:
+        """
+        Calculates every BC version in given RN dir and maps it to text if exists.
```

Comment:
Worth adding a comment about the json files - that the new practice about BC would be adding that json file to the RN dir.